crates/kubelet/src/kubelet.rs

```diff
@@ -265,102 +279,122 @@ pub trait Provider {
-                                }
-                            }
-                            // Secrets
-                            if let Some(seckey) = env_src.secret_key_ref.as_ref() {
-                                let name = seckey.name.as_deref().unwrap_or_default();
-                                match Api::v1Secret(client).within(ns).get(name) {
-                                    Ok(secret) => {
-                                        // I am not totally clear on what the outcome should
-                                        // be of a cfgmap key miss. So for now just return an
-                                        // empty default.
-
-                                        return secret
-                                            .stringData
-                                            .get(seckey.key.as_str())
-                                            .cloned()
-                                            .unwrap_or_default();
-                                    }
-                                    Err(e) => {
-                                        error!("Error fetching config map {}: {}", name, e);
-                                        return "".to_string();
-                                    }
-                                }
-                            }
-                            // Downward API (Field Refs)
-                            if let Some(cfkey) = env_src.field_ref.as_ref() {
-                                return fields
-                                    .get(cfkey.field_path.as_str())
-                                    .cloned()
-                                    .unwrap_or_default();
-                            }
-                            // Reource Fields (Not implementable just yet... need more of a model.)
-                        }
-                        "".to_string()
-                    }),
-                );
-            });
+            .and_then(|s| s.namespace.as_deref())
+            .unwrap_or("default");
+        for env_var in container.env.as_ref().unwrap_or_else(|| &empty).iter() {
+            let key = env_var.name.clone();
+            let value = match env_var.value.clone() {
+                Some(v) => v,
+                None => on_missing_value(client.clone(), env_var, ns, &field_map(pod)).await,
+            };
+            env.insert(key, value);
+        }
         env
     }
 }
 
+async fn on_missing_value(
+    client: APIClient,
+    env_var: &EnvVar,
+    ns: &str,
+    fields: &HashMap<String, String>,
+) -> String {
+    if let Some(env_src) = env_var.value_from.as_ref() {
+        // ConfigMaps
+        if let Some(cfkey) = env_src.config_map_key_ref.as_ref() {
+            let name = cfkey.name.as_deref().unwrap_or("");
+            match Api::<ConfigMap>::namespaced(client, ns).get(name).await {
+                Ok(cfgmap) => {
+                    // I am not totally clear on what the outcome should
```

Comment:
Note for the future: I think you are supposed to error if this happens if I remember correctly